### PR TITLE
New: AI Builder -  URL to Elementor [ED-12433] [ARPU-921] [ED-12664]

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: modules/ai/assets/js/editor/layout-module.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ export default class Module extends elementorModules.editor.utils.Module {
 				} ];
 
 				renderLayoutApp( {
+					mode: 'variation',
```
Comment (matipojo) [inline modules/ai/assets/js/editor/layout-module.js:43]: const

File: modules/ai/assets/js/editor/layout-app.js [hunk 2]
```diff
@@ -38,6 +40,7 @@ const LayoutApp = ( {
 };
 
 LayoutApp.propTypes = {
+	mode: PropTypes.oneOf( [ 'layout', 'variation' ] ),
```
Comment (matipojo) [inline modules/ai/assets/js/editor/layout-app.js:43]: const

File: modules/ai/assets/js/editor/pages/form-layout/components/attachments.js [hunk 3]
```diff
@@ -1,23 +1,41 @@
...
+				type: ATTACHMENT_TYPE_URL,
+			} ] }
+		/>
+	);
 };
 
 Attachments.propTypes = {
 	attachments: PropTypes.arrayOf( AttachmentPropType ).isRequired,
 	disabled: PropTypes.bool,
+	onAttach: PropTypes.func,
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/attachments.js:38]: required

File: modules/ai/assets/js/editor/pages/form-layout/components/attachments/attach-dialog.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+				onAttach={ props.onAttach }
+				onClose={ props.onClose }
+			/>;
+	}
+
+	return null;
+};
+
+AttachDialog.propTypes = {
+	type: PropTypes.string,
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/attachments/attach-dialog.js:21]: required

File: modules/ai/assets/js/editor/pages/form-layout/components/attachments/menu-item.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+			</Box>
+			<Typography>
+				{ props.title }
+			</Typography>
+		</Stack>
+	);
+};
+
+MenuItem.propTypes = {
+	title: PropTypes.string,
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/attachments/menu-item.js:37]: required

File: modules/ai/assets/js/editor/pages/form-layout/components/attachments/menu.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+					setSelectedType( null );
+				} }
+			/>
+		</>
+	);
+};
+
+Menu.propTypes = {
+	disabled: PropTypes.bool,
+	onAttach: PropTypes.func,
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/attachments/menu.js:66]: required

File: modules/ai/assets/js/editor/pages/form-layout/components/attachments/url-dialog.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+				},
+			} }
+		>
+			<DialogContent
+				sx={ {
+					padding: 0,
+				} }
+			>
+				{
+					isTimeout && <AlertDialog
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/attachments/url-dialog.js:70]: Extract

File: modules/ai/assets/js/editor/pages/form-layout/context/config.js [hunk 8]
```diff
@@ -24,6 +25,7 @@ export const ConfigProvider = ( props ) => {
 };
 
 ConfigProvider.propTypes = {
+	mode: PropTypes.oneOf( [ 'layout', 'variation' ] ).isRequired,
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/context/config.js:28]: const

File: modules/ai/assets/js/editor/pages/form-layout/hooks/use-attach-url-service.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+export const useAttachUrlService = ( ) => {
+	const [ currentUrl, setCurrentUrl ] = useState( '' );
+	const theme = useTheme();
+	const APP_BASE_URL = ElementorAiConfig.url_to_container_url;
+
+	const urlObject = new URL( APP_BASE_URL );
+	urlObject.searchParams.append( 'colorScheme', theme.palette.mode );
+	urlObject.searchParams.append( 'isRTL', 'rtl' === theme.direction ? 'true' : 'false' );
+	/* TODO: Add locale support */
+	urlObject.searchParams.append( 'locale', theme.locale );
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/hooks/use-attach-url-service.js:13]: Remove

File: modules/ai/assets/js/editor/utils/editor-integration.js [hunk 10]
```diff
@@ -30,7 +30,20 @@ export const getUiConfig = () => {
 	};
 };
 
+const REFORMAT_PROMPTS = [
```
Comment (matipojo) [inline modules/ai/assets/js/editor/utils/editor-integration.js:33]:
```suggestion
const VARIATIONS_PROMPTS = [
```

File: modules/ai/module.php [hunk 11]
```diff
@@ -113,9 +115,9 @@ private function register_layout_experiment() {
 		Plugin::$instance->experiments->add_feature( [
 			'name' => static::LAYOUT_EXPERIMENT,
 			'title' => esc_html__( 'Build with AI', 'elementor' ),
+			'description' => esc_html__( 'Tap into the potential of AI to easily create and customize containers to your specifications, right within Elementor. This feature comes packed with handy AI tools, including generation, variations, and URL references.', 'elementor' ),
 			'default' => Experiments_Manager::STATE_INACTIVE,
 			'status' => Experiments_Manager::RELEASE_STATUS_ALPHA,
```
Comment (matipojo) [inline modules/ai/module.php:120]:
```suggestion
			'default' => Experiments_Manager::STATE_INACTIVE,
			'status' => Experiments_Manager::RELEASE_STATUS_ALPHA,
			'hidden' => true,
```

**Review by blackcolt — APPROVED** [review]

**Pull request merged**